tools/wpt/browser.py

```diff
@@ -560,8 +560,15 @@ def install(self, dest=None, channel=None):
         os.remove(installer_path)
         return self.find_nightly_binary(dest)
 
-    def install_mojojs(self, dest):
-        url = self._latest_chromium_snapshot_url() + "mojojs.zip"
+    def install_mojojs(self, dest, channel, browser_binary):
```

Comment:
Should this check if the mojojs bindings already exist before trying to download them?

Reply:
We could, but we'd need to create a local lock file to record the version of the downloaded `mojojs.zip`; and it would only save us ~200KiB. WDYT?

Reply:
Oh for some reason I assumed the unzipped contents would be in a version-specific folder, and we could just check for that.

I'd still like to see us do this, but I'm ok to see it done in a future CL.

Reply:
> I'd still like to see us do this, but I'm ok to see it done in a future CL.

I just realized that this will happen for all tests that anyone runs on Chrome, rather than just mojojs tests (right?). Assuming I'm right, I will double down to I think we **should** do such caching... but again I'm ok with landing this as long as the follow-up happens :)

Reply:
Done. PTAL.

Reply:
Ok, this looks fine to me. So if someone goes between channels (or Chrome updates) they will re-download, but thats probably rare and thus fine.

Reply:
Yeah. If I'm reading the source code correctly, the seed for message scrambling is the full version string; if so, then one doesn't need to redownload if they already have the exact same version of `mojojs.zip` regardless of the builder or channel. But that's a big if, and it also depends on an implementation detail, so I decided to be safe here.